### PR TITLE
Mock Backend/API

### LINKED ISSUE #5: Mock backend
We'll use a json file that contains the three endpoints that we're interested in. Netlify, Auth0, and Twilio.
We can have a separate json file that has the current status saved for each endpoint in the feed.

### Storing current status
```json
{
  "netlify": {
     "timestamp": "<Date.now()>",
     "healthy": true
  }
}
```

Later on, we can think about mocking history like so:

### Storing each status:
```json
{
  "netlify": [
     {"healthy": true, "timestamp": "14:55"},
     {"healthy": false, "timestamp": "15:00"},
     {"healthy": false, "timestamp": "15:05"},
     {"healthy": true, "timestamp": "15:10"},
  ]
}
```

For now, the expected outcome is an `api/components/[name]` route that will serve these statuses.

The API should support GET (to get the status) and PUT (to update the status).

For this milestone, we will stick to current status only.

### PR DESCRIPTION
**Pull request by David-MacD**
Closes #5

- **create json backend add get endppoint**
- **PUT init**
- **Fix services endpoint**
- **Fix services/[service] GET and PUT endpoints**
- **Parameterise the json location in the services route**
- **Add content-type headers to all responses from the endpoints**
- **Add the data dir to .gitignore**
- **Remove bad export from service route**

### CODE REVIEW

File: src/lib/util.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export type Validator<T> = (o: any) => o is T;
```
Comment (Satty90) [inline src/lib/util.ts:1]: change o to obj please

File: src/app/api/services/route.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+} from "@/lib/util";
+
+const jsonLocation = "./data/mockBackend.json";
+const headers = { "Content-Type": "application/json" };
+
+export const GET = async (req: Request) => {
+  try {
+    const data = await fs.readFile(jsonLocation, "utf-8");
+    const services = tryParseJson<Service[]>(data, isServicesArray);
+    if (services) {
```
Comment (Satty90) [inline src/app/api/services/route.ts:16]: `if(services === null)`

File: src/app/api/services/route.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      return new Response(makeResponseBody("ok", services), { headers });
```
Comment (Satty90) [inline src/app/api/services/route.ts:17]: return the error path

File: src/app/api/services/[service]/route.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      return new Response(
+        makeResponseBody(
+          "error",
+          `There was a problem parsing the services data`,
+        ),
+        { headers },
+      );
+    }
+
+    const serviceIndexToUpdate = services.findIndex((s) =>
```
Comment (Satty90) [inline src/app/api/services/[service]/route.ts:68]: find a better name for s

File: src/app/api/services/[service]/route.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+const headers = { "Content-Type": "application/json" };
+
+export const GET = async (
+  _: Request,
+  { params }: ReqOptions,
+) => {
+  try {
+    const data = await fs.readFile(jsonLocation, "utf-8");
+    const services = tryParseJson<Service[]>(data, isServicesArray);
+    const service = services?.find((s) => s.name === params.service);
```
Comment (Satty90) [inline src/app/api/services/[service]/route.ts:23]: find a better name for s

File: src/app/api/services/[service]/route.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    if (service) {
+      return new Response(makeResponseBody("ok", service), { headers });
+    }
+    return new Response(
+      makeResponseBody(
+        "not found",
+        `Could not find the sevice called ${params.service}`,
```
Comment (Satty90) [inline src/app/api/services/[service]/route.ts:30]: typo service

File: src/app/api/services/route.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import fs from "node:fs/promises";
+import {
+  isServicesArray,
+  makeResponseBody,
+  type Service,
+  tryParseJson,
+} from "@/lib/util";
+
+const jsonLocation = "./data/mockBackend.json";
```
Comment (Satty90) [inline src/app/api/services/route.ts:9]: Change jsonLocation to seeder

**Review by Satty90 — CHANGES_REQUESTED** [review]
please see comments

File: src/app/api/services/route.ts (hunk 7 above)
Comment (David-MacD) [reply src/app/api/services/route.ts:9]: It's not a seeder. It's the file path.

**Review by Satty90 — APPROVED** [review]
LGTM

**Pull request merged**